File: Add_Border.py
# Arcade > Intro > 15
# Given a rectangular matrix of characters, add a border of asterisks(*) to it
# Ex input : picture = ["abc",
#                       "ded"]
#                   => ["*****",
#                       "*abc*",
#                       "*ded*",
#                       "*****"]

def addBorder(picture):
    m = []
    r = len(picture)+2
    w = len(picture[0])+2
    for row in range(r):
        if row == 0 or row == r-1:
            m.append("*"*w)
        else:
            m.append("*" + picture[row-1] + "*")
    return m

File: CodeSignalJS/add_Border.ts
// Arcade > Intro > 15
// Given a rectangular matrix of characters, add a border of asterisks(*) to it
// Ex input : picture = ["abc",
//                       "ded"]
//                   => ["*****",
//                       "*abc*",
//                       "*ded*",
//                       "*****"]

function addBorder(p: string[]) {
  p = p.map(x=>"*"+x+"*")
  p.unshift(("*").repeat(p[0].length));
  p.push(("*").repeat(p[0].length));
  return p
}

module.exports = addBorder;